src/audio/module_adapter/module/volume/volume.c

```diff
@@ -1169,6 +1170,14 @@ static int volume_process(struct processing_module *mod,
 
 		avail_frames -= frames;
 	}
+#if CONFIG_COMP_PEAK_VOL
+	cd->peak_cnt++;
```

Comment:
should use frames to calculate time, we can't assume volume_process handle 1ms per time.

Reply:
mod->dev->period/1000 is the ms

Reply:
I don't think this works with many instances and different scheduling. You need to calculate the number of copy() operations to per update for every peakvolume instance.

Reply:
So, please restore kconfig to ms instead of periods.

Reply:
cd->peak_cnt += mod->dev->period/1000；
or
if (cd->peak_cnt == CONFIG_PEAK_METER_UPDATE_PERIOD*1000/mod->dev->period)

Reply:
I'd like to move the count calculation to prepare function. Done thanks